components/encryption/src/config.rs

```diff
@@ -55,8 +50,8 @@ pub struct FileConfig {
 pub struct KmsConfig {
     pub key_id: String,
 
-    // Providing access_key and secret_access_key is recommended as it has
-    // security risk.
+    // It is preferable to use the cloud provider SDK to automatically rotate credentials
+    // Using access_key and secret_access_key creates a security risk.
```

Comment:
yes, do you mind help me update the macro?

Reply:
I think it would be better to make this change separately. I have experience doing the same thing in our go code and always found that the solution to these issues is just to delete any custom aws credentials code and use what the SDK provides by default.

Reply:
"use what the SDK provides by default` is implemented in the `new_client!` macro. We just need a variant of the macro that don't look at the custom credential but just use the default credentials.

Reply:
the explicit credentials is not being used by anyone for the KMS flow. It is only used for S3 for BR.

Reply:
I think we are in agreement. I will make a separate PR for this though.

Reply:
okay